packages/react-reconciler/src/ReactFiberNewContext.js

```diff
@@ -641,6 +663,28 @@ function propagateParentContextChanges(
   workInProgress.flags |= DidPropagateContext;
 }
 
+function checkIfComparedContextValuesChanged(
+  oldComparedValue: mixed,
+  newComparedValue: mixed,
+): boolean {
+  if (isArray(oldComparedValue) && isArray(newComparedValue)) {
```

Comment:
this needs to check the array length, right? then you won't need to compare `i` against both lengths in the for test condition